CMakeLists.txt

```diff
@@ -688,6 +722,28 @@ if(WITH_OPTIM)
                 add_feature_info(PCLMUL_CRC 1 "Support CRC hash generation using PCLMULQDQ, using \"${PCLMULFLAG} ${SSE4FLAG}\"")
             endif()
         endif()
+    elseif(BASEARCH_PPC_FOUND)
+        if(HAVE_POWER8 OR HAVE_POWER9)
+            add_definitions(-DPOWER_FEATURES)
+            list(APPEND ZLIB_ARCH_HDRS ${ARCHDIR}/power.h)
+            list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/power.c)
+            if(WITH_POWER8 AND HAVE_POWER8)
+                add_definitions(-DPOWER8)
+                set(ZLIB_POWER8_SRCS
+                    ${ARCHDIR}/slide_hash_power8.c)
+                set_source_files_properties(
+                    ${ZLIB_POWER8_SRCS}
+                    PROPERTIES COMPILE_FLAGS ${POWER8FLAG})
+            endif()
+            if(WITH_POWER9 AND HAVE_POWER9)
+                add_definitions(-DPOWER9)
+                set(ZLIB_POWER9_SRCS )
```

Comment:
This looks incomplete...

Reply:
@mtl1979 that's on purpose actually. The PR does not contain any optimizations that require POWER9-only instructions (but others will). So this is left empty for now to be used in the future. I can add a comment to that line if you like.

Reply:
It might be better to just remove the block and add it when there is files that actually use POWER9 instructions...

Reply:
Ok, so I'll remove all mentions to P9 from `CMakeLists.txt`, `configure`, and `arch/power/power.c` and queue them for a future PR.

Reply:
The POWER9 stuff does look good but I agree it was kind of confusing as to why they were there and I was trying to figure out where they were being used.

Reply:
I still haven't made the change. So you prefer it gone for now or I can leave it there? It does offer the whole scaffolding for adding a P9 optimized function (like madler/zlib#459), but it is not being exercised in this PR specifically.

Reply:
Having the half-finished code confuses a lot... Either finish the Power9 optimized code or remove the unneeded lines mentioning ZLIB_POWER9_SRCS.

Reply:
I suggest removing it for now. It will likely reduce the size of the PR and probably lead to it getting merged quicker by @Dead2. Then just add it back in a PR that uses Power9. It is a pain I know.

Reply:
OK, I've just cleaned up the code.